sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodShouldBeNamedAccordingToSynchronicity.CSharp8.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics
+{
+    public abstract class IEnumerableReproducer
+    {
+        protected abstract IAsyncEnumerable<int> GetFilesAsync(string folderName); // Compliant (https://github.com/SonarSource/sonar-dotnet/issues/3433)
```

Comment:
maybe also add a class that implements `IAsyncEnumerable` ?  and make sure that works?

Reply:
Good catch. This worked fine for types which derived from tasks since `.DerivesFromAny(KnownType)` was used but `IAsyncEnumerable` is an interface..